feat(@desktop:communities): Minting functionality with dummy data

endulab opened the pull request:
Add Community Tokens testing UI with minting button, enabled by a Advanced Settings toggle. Add minting module, view and needed models.
Add community_tokens service to call collectibles smart contract functions.

Issue #8921

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 240e4106 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/tests-nim/job/PR-9082/1/) | 2023-01-12 13:45:15 | ~2 min | `tests-nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/tests-nim/job/PR-9082/1/consoleText) |

endulab commented:
@noeliaSD @jrainville because I added a lot of improvements I changed this pr to a draft pr.
It will be evolving and I will switch to pr when become fully functional.

I added community_tokens nim service, backend function, authentication and toggle in Advanced section.

noeliaSD commented on src/app/modules/main/communities/minting/models/token_item.nim:
```diff
@@ -0,0 +1,100 @@
+    name: string
+    description: string
+    icon: string
+    supply: string
```
Could the supply be different than an int??

endulab replied:
Changed to int

noeliaSD commented on src/app/modules/main/communities/minting/models/token_item.nim:
```diff
@@ -0,0 +1,100 @@
+): TokenItem =
+  result = TokenItem()
+  result.tokenType = TokenType.ERC721
+  result.tokenAddress = description
```
??
```suggestion
  result.tokenAddress = tokenAddress
```

endulab replied:
fixed

noeliaSD commented on src/app_service/service/community_tokens/dto/deployment_parameters.nim:
```diff
@@ -0,0 +1,24 @@
+  DeploymentParameters* = object
+    name*: string
+    symbol*: string
+    supply*: string
```
??
```suggestion
    supply*: int
```

noeliaSD reviewed (commented):
Great!! Just some minor comments!

endulab commented:
@noeliaSD I changed supply type to int. It was string first because of conversions issues.
In smart contracts this number can be very big and we should have used uint256 for this.
However, it turned out that the max supply value can be 999999999, so even int32 can be used to hold this,
and we does not need to worry about conversions.

noeliaSD reviewed (approved):
LGTM! :)

jrainville commented on src/app/modules/main/communities/minting/models/token_item.nim:
```diff
@@ -0,0 +1,100 @@
+  return self.icon
+
+proc getSupply*(self: TokenItem): int =
+  return self.supply
```
Any chance of this being a very big number? If so, we should pass it as a string and use some BIgInt lib

endulab replied:
I had string here. However you can see: https://github.com/status-im/status-desktop/pull/9082#issuecomment-1403806397. Because our limit is 999999999 we can use int for now.

endulab commented on ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokenPanel.qml:
```diff
@@ -0,0 +1,102 @@
+
+    property var tokensModel
+
+    signal mintCollectible(string address, string name, string symbol, string description, string supply,
```
Fixed
